src/spatial/2d.js

```diff
@@ -929,6 +834,14 @@ Crafty.c("2D", {
             }
             this.trigger("Move", old);
 
+        } else if (name === '_z') {
+            var intValue = value <<0;
+            value = value==intValue ? intValue : intValue+1;
+            this._globalZ = value*100000+this[0]; //magic number 10^5 is the max num of entities
+            this[name] = value;
+            this.trigger("Reorder");
+            this.trigger("Invalidate");
+            return;
         }
```

Comment:
Nope, because we return early in this branch.

IIRC that's because we want "Reorder" to trigger after the value has been set.